### ISSUE TITLE
Use public interface in tests wherever possible (don't use :::)

### ISSUE THREAD

**Issue by MichaelChirico** [r-lib/lintr#1692]
e.g. this suite can be replaced by `expect_lint()` tests:

https://github.com/r-lib/lintr/blob/bed54c55523e96f7c89154ae31702cd74bb209d8/tests/testthat/test-implicit_integer_linter.R#L2-L35

Here's a list of tests using `:::`, generated with `grep -Flr ":::" tests/testthat | awk -F/ '{print "- [ ] " $3}' | sort`, we can cross some off if they are just used for coverage & testing with public API is difficult:

- [ ] test-absolute_path_linter.R #2801
- [ ] test-cache.R #2801
- [x] test-ci.R
- [x] test-commented_code_linter.R #1726
- [x] test-comments.R 
- [x] ~test-dir_linters.R~
- [x] test-error.R
- [ ] test-exclusions.R #1720
- [ ] test-get_source_expressions.R #2805
- [x] test-implicit_integer_linter.R #1693 
- [ ] test-linter_tags.R #2801
- [ ] test-lint_dir.R #2801
- [x] test-lint_package.R
- [ ] test-make_linter_from_regex.R #2801
- [ ] test-methods.R #2801
- [x] test-namespace_linter.R
- [x] test-namespace.R #1721
- [ ] test-normalize_exclusions.R #2801
- [ ] test-object_name_linter.R #2801
- [ ] test-parse_exclusions.R #2801
- [ ] test-settings.R #2801
- [x] test-undesirable_operator_linter.R
- [x] test-use_lintr.R #1774
- [x] test-utils.R #1720

**Comment by IndrajeetPatil** [comment]
I think `test-namespace_linter.R` is a false positive here. It's picking up `:::` from tests where they are deliberate. E.g.

```r
  expect_lint(
    "stats:::sdd(c(1,2,3))",
    list(message = rex::rex("'sdd' does not exist in {stats}")),
    linter
  )
```
